lib/src/stream_extensions.dart

```diff
@@ -54,4 +54,27 @@ extension StreamExtensions<T> on Stream<T> {
     });
     return completer.future;
   }
+
+  /// Eagerly listens to this stream and buffers events until needed.
+  ///
+  /// The returned stream will emit the same events as this stream, starting
+  /// from when this method is called. The events are delayed until the returned
+  /// stream is listened to, at which point all buffered events will be emitted
+  /// in order, and then further events from this stream will be emitted as they
+  /// arrive.
+  ///
+  /// The buffer will retain all events until the returned stream is listened
+  /// to, so if the stream can emit arbitrary amounts of data callers should be
+  /// careful to listen to it eventually or call `stream.listen(null).cancel()`
+  /// to discard it if it's not needed.
```

Comment:
Consider changing some (or all) of he "it"s in this sentence with what they refer to.

```dart
  /// careful to listen to the returned stream eventually, 
  /// or do `returnedStream.listen(null).cancel()` to discard the buffered data 
  /// if it becomes clear that the data isn't needed.
```
(Avoiding "it" almost always improves documentation.)